File: go/0011-container-with-most-water.go
func Min(x, y int) int {
    if x < y{
        return x
    }
    return y
}

func maxArea(height []int) int {
    var left, maxWater, tempWater int
    right := len(height)-1
    
    for left<right {
        tempWater = Min(height[right],height[left])*(right-left)
        if tempWater > maxWater {
            maxWater = tempWater
        }
        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }
    return maxWater
}

File: python/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxVal = 0
        left = 0
        right = len(height)-1

        while left < right:
            val = (min(height[left],height[right]) * (right-left))
            maxVal = max(maxVal, val)
            
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return maxVal